packages/agreement-process/src/services/agreementSubmissionProcessor.ts

```diff
@@ -0,0 +1,349 @@
+  const updateSeed = getUpdateSeed(
+    descriptor,
+    consumer,
+    eService,
+    agreement,
+    payload,
+    stamps,
+    newState,
+    suspendedByPlatform
+  );
+
+  const updatedAgreement = {
+    ...agreement,
+    ...updateSeed,
+  };
+
+  const updatedAgreementEvent = toCreateEventAgreementUpdated(
+    updatedAgreement,
+    agreementData.metadata.version
+  );
+
+  const agreements = (
+    await agreementQuery.getAgreements({
+      producerId: agreement.producerId,
+      consumerId: agreement.consumerId,
+      eserviceId: agreement.eserviceId,
+      agreementStates: [agreementState.active, agreementState.suspended],
+    })
+  ).filter((a: WithMetadata<Agreement>) => a.data.id !== agreement.id);
+
+  const relatedAgreementUpdates: Array<CreateEvent<AgreementEvent>> =
+    isActiveOrSuspended(newState)
+      ? await Promise.all(
+          agreements.map(
+            async (
+              agreement: WithMetadata<Agreement>
+            ): Promise<CreateEvent<AgreementEvent>> => {
+              const updateSeed: UpdateAgreementSeed = {
+                state: agreementState.archived,
+                certifiedAttributes: agreement.data.certifiedAttributes.map(
+                  (ca) => ({
+                    type: agreementAttributeType.CERTIFIED,
+                    id: ca.id,
+                  })
+                ),
+                declaredAttributes: agreement.data.declaredAttributes.map(
+                  (da) => ({
+                    type: agreementAttributeType.DECLARED,
+                    id: da.id,
+                  })
+                ),
+                verifiedAttributes: agreement.data.verifiedAttributes.map(
+                  (va) => ({
+                    type: agreementAttributeType.VERIFIED,
+                    id: va.id,
+                  })
+                ),
+                stamps: {
+                  ...agreement.data.stamps,
+                  archiving: {
+                    who: authData.userId,
+                    when: utcToZonedTime(new Date(), "Etc/UTC"),
+                  },
+                },
+              };
+
+              return toCreateEventAgreementUpdated(
+                {
+                  ...agreement.data,
+                  ...updateSeed,
+                },
+                agreement.metadata.version
+              );
+            }
+          )
+        )
+      : [];
+
+  const createContractEvents: Array<CreateEvent<AgreementEvent>> =
+    updatedAgreement.state === agreementState.active && agreements.length === 0
+      ? [
+          await createContract(
+            updatedAgreement,
+            updatedAgreementEvent.version + 1,
+            eService,
+            consumer,
+            updateSeed,
+            tenantQuery,
+            constractBuilder
+          ),
+        ]
+      : [];
+
+  return [
+    updatedAgreementEvent,
+    ...relatedAgreementUpdates,
+    ...createContractEvents,
+  ];
```

Comment:
In the last commit I have changed the way with event's array was built, `createContractEvents` was aggregated with others and processed in correctly order.
cc @MalpenZibo